### PR TITLE
ref(ui) Fix more URLs for customer domains

### PR DESCRIPTION
**Pull request by markstory**
* Fix URL generation in monitors. This was a more invasive change as the routes won't receive `orgId` in the future due to how organization slugs are propagated to class components.
* Updated URL normalization for Organization general settings.

### CODE REVIEW

File: static/app/views/monitors/create.tsx [hunk 1]
```diff
@@ -11,12 +14,22 @@ import {Monitor} from './types';
 type Props = AsyncView['props'] & RouteComponentProps<{orgId: string}, {}>;
 
 export default class CreateMonitor extends AsyncView<Props, AsyncView['state']> {
+  static contextTypes = {
+    router: PropTypes.object,
```
Comment (dashed) [inline static/app/views/monitors/create.tsx:18]: This router context doesn't seem to be used anywhere.

File: static/app/views/monitors/details.tsx [hunk 2]
```diff
@@ -25,16 +27,25 @@ type State = AsyncView['state'] & {
 };
 
 class MonitorDetails extends AsyncView<Props, State> {
+  static contextTypes = {
+    router: PropTypes.object,
```
Comment (dashed) [inline static/app/views/monitors/details.tsx:31]: This router context doesn't seem to be used anywhere.

**Review by dashed — APPROVED** [review]

File: static/app/views/monitors/create.tsx (hunk 1 above)
Comment (markstory) [reply static/app/views/monitors/create.tsx:18]: It isn't used here, but parts of AsyncComponent use `router` so it needs to be here too.

File: static/app/views/monitors/details.tsx (hunk 2 above)
Comment (markstory) [reply static/app/views/monitors/details.tsx:31]: Yeah, it is used by AsyncView/AsyncComponent though.

**Pull request merged**